prototype2/sonde/SoNDeBase.cpp

```diff
@@ -115,6 +128,13 @@ void SONDEIDEABase::processing_thread() {
         mystats.kafka_dr_errors = eventprod.stats.dr_errors;
         mystats.kafka_dr_noerrors = eventprod.stats.dr_noerrors;
         produce_timer.now();
+
+        if (!histograms.isEmpty()) {
+          XTRACE(PROCESS, DEB, "Sending histogram for %zu readouts",
+                 histograms.hit_count());
+          histfb.produce(histograms);
```

Comment:
From what I understood, the producer only sends histograms during idle processing cycles. I guess this works fine, but is there not a risk that under a heavy processing workload no histogram updates get sent?

Reply:
True, this is my general philosophy: don't lose data over trivial tasks. The stats are not lost, just posted a little later. In general we really should not run into this situation, so either way would be good.

Reply:
If we did it the other way around there is a slight risk of losing packets because the fifo fills up while posting the stats. But this is probably more philosophical anyway :-)